src/new_rdata/basic.rs

```diff
@@ -0,0 +1,531 @@
+    ) -> Result<Self, ParseError> {
+        N::parse_from_message(message, start).map(|name| Self { name })
+    }
+}
+
+//--- Building into DNS messages
+
+impl<N: ?Sized + BuildIntoMessage> BuildIntoMessage for Ptr<N> {
+    fn build_into_message(&self, builder: build::Builder<'_>) -> BuildResult {
+        self.name.build_into_message(builder)
+    }
+}
+
+//----------- HInfo ----------------------------------------------------------
+
+/// Information about the host computer.
+#[derive(Clone, Debug, PartialEq, Eq, BuildBytes, ParseBytes, SplitBytes)]
+pub struct HInfo<'a> {
+    /// The CPU type.
+    pub cpu: &'a CharStr,
+
+    /// The OS type.
+    pub os: &'a CharStr,
+}
+
+//--- Parsing from DNS messages
+
+impl<'a> ParseFromMessage<'a> for HInfo<'a> {
+    fn parse_from_message(
+        message: &'a Message,
+        start: usize,
+    ) -> Result<Self, ParseError> {
+        message
+            .contents
+            .get(start..)
+            .ok_or(ParseError)
+            .and_then(Self::parse_bytes)
+    }
+}
+
+//--- Building into DNS messages
+
+impl BuildIntoMessage for HInfo<'_> {
+    fn build_into_message(
+        &self,
+        mut builder: build::Builder<'_>,
+    ) -> BuildResult {
+        self.cpu.build_into_message(builder.delegate())?;
+        self.os.build_into_message(builder.delegate())?;
+        Ok(builder.commit())
+    }
+}
+
+//----------- Mx -------------------------------------------------------------
+
+/// A host that can exchange mail for this domain.
+#[derive(
+    Copy,
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    BuildBytes,
+    ParseBytes,
+    SplitBytes,
+)]
+#[repr(C)]
+pub struct Mx<N: ?Sized> {
+    /// The preference for this host over others.
+    pub preference: U16,
+
+    /// The domain name of the mail exchanger.
+    pub exchange: N,
+}
+
+//--- Parsing from DNS messages
+
+impl<'a, N: ParseFromMessage<'a>> ParseFromMessage<'a> for Mx<N> {
+    fn parse_from_message(
+        message: &'a Message,
+        start: usize,
+    ) -> Result<Self, ParseError> {
+        let (&preference, rest) = <&U16>::split_from_message(message, start)?;
+        let exchange = N::parse_from_message(message, rest)?;
+        Ok(Self {
+            preference,
+            exchange,
+        })
+    }
+}
+
+//--- Building into DNS messages
+
+impl<N: ?Sized + BuildIntoMessage> BuildIntoMessage for Mx<N> {
+    fn build_into_message(
+        &self,
+        mut builder: build::Builder<'_>,
+    ) -> BuildResult {
+        builder.append_bytes(self.preference.as_bytes())?;
+        self.exchange.build_into_message(builder.delegate())?;
+        Ok(builder.commit())
+    }
+}
+
+//----------- Txt ------------------------------------------------------------
+
+/// Free-form text strings about this domain.
+#[derive(AsBytes, BuildBytes)]
+#[repr(transparent)]
+pub struct Txt {
+    /// The text strings, as concatenated [`CharStr`]s.
+    content: [u8],
```

Comment:
This seems wrong. A txt record is a collection of strings. Boundaries need to be preserved.

Reply:
Ah. Maybe an extra comment with what you just wrote.